### PR TITLE
feat(subscriptions): Delete subscriptions from snuba if they don't exist in sentry

### PR DESCRIPTION
**Pull request by wedamija**
If we receive an update from a subscription that doesn't have a `QuerySubscription` in sentry then
delete it from snuba. This should help correct situations where we have subscriptions that are no
longer valid from taking up resources.

### CODE REVIEW

File: src/sentry/snuba/query_subscription_consumer.py [hunk 1]
```diff
@@ -166,6 +169,9 @@ def handle_message(self, message):
                         "value": message.value(),
                     },
                 )
+                _delete_from_snuba(
+                    self.topic_to_dataset[message.topic()], contents["subscription_id"]
```
Comment (markstory) [inline src/sentry/snuba/query_subscription_consumer.py]: Could there be other remnants in the database for the alert rule or rule triggers that also need to be cleaned up?

Comment (wedamija) [reply src/sentry/snuba/query_subscription_consumer.py]: If there are then there's no way for us to find them without a `QuerySubscription`. This shouldn't be the case in general due to how we handle deletes - this is mostly to handle the case where we retry POST and create multiple subscriptions, and also if there are any problems when updating an alert rule.

File: src/sentry/snuba/subscriptions.py [hunk 2]
```diff
@@ -187,7 +190,10 @@ def delete_snuba_subscription(subscription):
         # TODO: Move this call to snuba into a task. This lets us successfully delete a
         # subscription in postgres and rollback as needed without having to create/delete
         # from snuba
-        _delete_from_snuba(subscription)
+        try:
+            _delete_from_snuba(QueryDatasets(subscription.dataset), subscription.subscription_id)
+        except Exception:
+            logger.exception("Failed to delete unused subscription from snuba.")
```
Comment (tkaemming) [inline src/sentry/snuba/subscriptions.py]: Would you want the subscription ID and dataset in this log line?

**Review by tkaemming — APPROVED** [review]

File: src/sentry/snuba/subscriptions.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/snuba/subscriptions.py]: Dang, I put this try/catch in the completely wrong place. Wanted to make sure failures wouldn't block the consumer

**Pull request merged**